lib/puppet/type/yumrepo.rb

```diff
@@ -324,7 +324,7 @@
     newvalues(%r{.*}, :absent)
     validate do |value|
       next if %r{^(absent|_none_)$}.match?(value.to_s)
-      next if value.to_s.empty? && Facter.value(:operatingsystemmajrelease).to_i >= 8
+      next if value.to_s.empty? && Facter.value('os.distro.release.major').to_i >= 8
```

Comment:
I'm not sure why there's both `os.distro.release` and `os.release`, but I think the former is preferred based on https://github.com/puppetlabs/puppet-lint/blob/d0e55603755b0e5eb63b8601d117f4628f925ed1/lib/puppet-lint/plugins/legacy_facts/legacy_facts.rb#L77?

```suggestion
      next if value.to_s.empty? && Facter.value('os.release.major').to_i >= 8
```